protected/views/patient/_patient_gp.php

```diff
@@ -73,7 +73,7 @@
             <div class="data-label">Referral:</div>
           </div>
           <div class="large-8 column">
-            <div class="data-value"><?php echo CHtml::link($this->patient->referral->file_name, array('downloadReferral', 'id' => $this->patient->referral->patient_id));?></div>
+            <div class="data-value"><?php echo $this->patient->referral->file_name ? CHtml::link($this->patient->referral->file_name, array('downloadReferral', 'id' => $this->patient->referral->patient_id)) : 'None'; ?></div>
```

Comment:
This doesn't raise any issues in PHPStorm (which surprises me) and is used elsewhere on the same page, so it's probably OK.